packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts

```diff
@@ -0,0 +1,140 @@
+import { ICheckboxStyles } from './Checkbox.Props';
+import {
+  ITheme,
+  mergeStyleSets,
+  getFocusStyleBasedOnAncestorClass
+} from '../../Styling';
+import { memoizeFunction } from '../../Utilities';
+
+const MS_CHECKBOX_LABEL_SIZE = '20px';
+const MS_CHECKBOX_TRANSITION_DURATION = '200ms';
+const MS_CHECKBOX_TRANSITION_TIMING = 'cubic-bezier(.4, 0, .23, 1)';
+const MS_CHECKBOX_HIGHCONTRAST_ICONBOXSIZE = '16px';
+
+const MS_HIGHCONTRAST_ACTIVE = '@media screen and (-ms-high-contrast: active)';
+const MS_HIGHCONTRAST_BLACK_ON_WHITE = '@media screen and (-ms-high-contrast: black-on-white)';
+
+export const getStyles = memoizeFunction((
+  theme: ITheme,
+  customStyles?: ICheckboxStyles
+): ICheckboxStyles => {
+  const { semanticColors, palette } = theme;
+  const checkmarkFontColor = semanticColors.bodyBackground;
+  const checkboxBorderColor = semanticColors.inputBorder;
+  const checkboxHoveredBorderColor = semanticColors.inputBorderHovered;
+  const checkboxCheckedBackground = semanticColors.inputBackgroundChecked;
+  const checkboxCheckedHoveredBackground = semanticColors.inputBackgroundCheckedHovered;
+  const checkboxDisabledBackground = semanticColors.disabledText;
+  const checkboxDisabledTextColor = semanticColors.disabledText;
+
+  const styles: ICheckboxStyles = {
+    root: {
+      overflow: 'hidden',
+      position: 'relative',
+    },
+    label: [
+      getFocusStyleBasedOnAncestorClass(theme, '.ms-Fabric.is-focusVisible .is-inFocus'),
+      {
+        display: 'inline-flex',
+        alignItems: 'center',
+        cursor: 'pointer',
+        position: 'relative',
+        margin: '0 -4px',
+        userSelect: 'none'
+      }
+    ],
+    labelReversed: {
+      flexDirection: 'row-reverse',
+      justifyContent: 'flex-end'
+    },
+    labelDisabled: {
+      cursor: 'default'

```

Comment:
This the way I see it. The base styles are for the default/enabled/unchecked state of a component.
The other styles are overrides, implicitly stating that the base style (ex: just 'label' not 'labelDisabled' ) should be applied first. I have done this in the Checkbox.tsx file in the _getClassNames() function. And because, we control how the styles are cascaded in the function, we (the library) can also control the order